src/main.rs

```diff
@@ -0,0 +1,173 @@
+use {
+    borsh::BorshDeserialize,
+    clap::Parser,
+    core::panic,
+    observation::{
+        Body,
+        SignedBody,
+    },
+    posted_message::PostedMessageUnreliableData,
+    secp256k1::SecretKey,
+    solana_account_decoder::UiAccountEncoding,
+    solana_client::{
+        nonblocking::pubsub_client::PubsubClient,
+        pubsub_client::PubsubClientError,
+        rpc_config::{
+            RpcAccountInfoConfig,
+            RpcProgramAccountsConfig,
+        },
+    },
+    solana_sdk::pubkey::Pubkey,
+    std::{
+        fs,
+        str::FromStr,
+        time::Duration,
+    },
+    tokio::time::sleep,
+    tokio_stream::StreamExt,
+};
+
+mod config;
+mod observation;
+mod posted_message;
+mod serde_array;
+
+const PYTHNET_CHAIN_ID: u16 = 26;
+
+struct ListenerConfig {
+    ws_url:              String,
+    secret_key:          SecretKey,
+    wormhole_pid:        Pubkey,
+    accumulator_address: Pubkey,
+}
+
+fn find_message_pda(wormhole_pid: &Pubkey, ring_index: u32) -> Pubkey {
+    Pubkey::find_program_address(
+        &[b"AccumulatorMessage", &ring_index.to_be_bytes()],
+        wormhole_pid,
+    )
+    .0
+}
+
+async fn run_listener(config: ListenerConfig) -> Result<(), PubsubClientError> {
+    let client = PubsubClient::new(config.ws_url.as_str()).await?;
+    let (mut stream, unsubscribe) = client
+        .program_subscribe(
+            &config.wormhole_pid,
+            Some(RpcProgramAccountsConfig {
+                filters:        None,
```

Comment:
you can use filter on the header of the account to ensure you only get the messages with the header we want (`msu`). it saves you some bandwidth as in each slot two accounts from wormhole get updates (one is the message, the other one is the sequence PDA for each emitter)